### PR TITLE
[6.2.z] qe_test_coverage 1374505 1447829

### PR DESCRIPTION
**Pull request by ldjebran**
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1374505
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1447829

```console
(sat-6.2.z) dlezz@elysion:~/projects/robottelo-fork$ py.test tests/foreman/ui/test_repository.py -v -k "test_positive_create_as_non_admin_user"
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.32, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.2.z/bin/python
cachedir: .cache
shared_function enabled - ON - scope:  - storage: file
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.15.0, services-1.1.14, cov-2.3.1
collected 55 items 
2017-05-11 14:24:16 - conftest - DEBUG - Found WONTFIX in decorated tests ['1110476', '1156555', '1269196', '1402826', '1245334', '1221971', '1217635', '1226425', '1199150', '1204686', '1267224', '1103157', '1230902', '1214312', '1079482']

2017-05-11 14:24:16 - conftest - DEBUG - Collected 55 test cases


tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_create_as_non_admin_user <- robottelo/decorators/__init__.py PASSED

================================================= 54 tests deselected ==================================================
======================================= 1 passed, 54 deselected in 70.78 seconds =======================================
```

### CODE REVIEW

File: tests/foreman/ui/test_repository.py [hunk 1]
```diff
@@ -257,6 +263,63 @@ def test_positive_create_repo_with_checksum(self):
...
+            # check administer->users page
+            with self.assertRaises(UINoSuchElementError):
+                session.nav.go_to_users()
+            # ensure that the created user has only the assigned
+            # permissions, check that hosts menu tab does not exist
+            self.assertIsNone(
+                self.content_views.wait_until_element(
+                    menu_locators['menu.hosts'], timeout=5)
+            )
+            self.products.search(product.name).click()
```
Comment (oshtaier) [inline tests/foreman/ui/test_repository.py]: `self.products.click(self.products.search(product.name))`

Comment (ldjebran) [inline tests/foreman/ui/test_repository.py]: will resolve as : self.products.search_and_click(product.name)

**Comment by ldjebran** [comment]
```console
(sat-6.2.z) dlezz@elysion:~/projects/robottelo-fork$ py.test tests/foreman/ui/test_repository.py -v -k "test_positive_create_as_non_admin_user"
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.32, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.2.z/bin/python
cachedir: .cache
shared_function enabled - ON - scope:  - storage: file
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.15.0, services-1.1.14, cov-2.3.1
collected 55 items 
2017-05-11 14:43:41 - conftest - DEBUG - Found WONTFIX in decorated tests ['1110476', '1156555', '1269196', '1402826', '1245334', '1221971', '1217635', '1226425', '1199150', '1204686', '1267224', '1103157', '1230902', '1214312', '1079482']

2017-05-11 14:43:41 - conftest - DEBUG - Collected 55 test cases


tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_create_as_non_admin_user <- robottelo/decorators/__init__.py PASSED

================================================= 54 tests deselected ==================================================
======================================= 1 passed, 54 deselected in 72.04 seconds =======================================
(sat-6.2.z) dlezz@elysion:~/projects/robottelo-fork$ 
```

**Comment by ldjebran** [comment]
@oshtaier comment addressed

**Comment by oshtaier** [comment]
Anyway, one of them is in 'NEW' state, so corresponding skip decorator should be present

**Comment by ldjebran** [comment]
@oshtaier the creator of the bug was aware of the original bug that was closed as CLOSED ERRATA, but one of the customers still had the same problem.
@JacobCallahan what do you think can we mark the second bug as duplicate in this case and reopen the original one? (please note that was not able to reproduce the bug on the latest 6.2.9)

**Comment by ldjebran** [comment]
@oshtaier, make a skip decorator even if the test is passing?

**Comment by oshtaier** [comment]
It can't pass if it is 'NEW' state, otherwise close the bug

**Comment by JacobCallahan** [comment]
I'd recommend making the new bug, the primary one. Then noting in the test that it also covers the original. This way we aren't testing the bug before it is ready and we still correctly show that we are covering both bugs with this test.

**Comment by ldjebran** [comment]
put it do not merge need more info on this bug as cannot reproduce, by test or manually

**Comment by ldjebran** [comment]
```console
(sat-6.2.z) dlezz@elysion:~/projects/robottelo-fork$ py.test tests/foreman/ui/test_repository.py -v -k "test_positive_create_as_non_admin_user"
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.32, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.2.z/bin/python
cachedir: .cache
shared_function enabled - ON - scope:  - storage: file
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.15.0, services-1.1.14, cov-2.3.1
collected 56 items 
2017-05-15 12:03:47 - conftest - DEBUG - Found WONTFIX in decorated tests ['1110476', '1156555', '1269196', '1402826', '1245334', '1221971', '1217635', '1226425', '1199150', '1204686', '1267224', '1103157', '1230902', '1214312', '1079482']

2017-05-15 12:03:47 - conftest - DEBUG - Collected 56 test cases


tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_create_as_non_admin_user <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_create_as_non_admin_user_with_cv_published <- robottelo/decorators/__init__.py SKIPPED

================================================= 54 tests deselected ==================================================
================================= 1 passed, 1 skipped, 54 deselected in 94.47 seconds ==================================
(sat-6.2.z) dlezz@elysion:~/projects/robottelo-fork$ 
```
the skipped test fail to found the new repo button as described in the bug
```console
Error
Traceback (most recent call last):
  File "/home/dlezz/projects/robottelo-fork/robottelo/decorators/__init__.py", line 255, in wrapper
    return func(*args, **kwargs)
  File "/home/dlezz/projects/robottelo-fork/tests/foreman/ui/test_repository.py", line 383, in test_positive_create_as_non_admin_user_cv_pubilished
    url=FAKE_1_YUM_REPO,
  File "/home/dlezz/projects/robottelo-fork/robottelo/ui/factory.py", line 233, in make_repository
    Repos(session.browser).create(**create_args)
  File "/home/dlezz/projects/robottelo-fork/robottelo/ui/repository.py", line 23, in create
    self.click(locators['repo.new'])
  File "/home/dlezz/projects/robottelo-fork/robottelo/ui/base.py", line 665, in click
    .format(type(self).__name__, target)
UINoSuchElementError: Repos: element ('xpath', "//button[contains(@ui-sref,'repositories.new')]") was not found while trying to click
```

**Comment by ldjebran** [comment]
@oshtaier , after investigations on the second bug a new test has been created, please review

**Pull request merged**